### DISCUSSION TITLE
Display boot menu on both framebuffer and text consoles?

### DISCUSSION THREAD

**Discussion by bkshoemaker** [zbm-dev/zfsbootmenu#685] [Questions] [answered]
I tried enabling text-mode serial output for ZBM via command line option, which displayed the boot menu fine* over serial, but that disabled the menu on the framebuffer console. Is there a way to make the menu display on both consoles at the same time?

*the size of the menu was actually severely constrained -- is there a way to set columns and rows for serial output from the command line?

**Comment by zdykstra** [comment, marked as answer]
The main interface can only be shown on one display at a time. It will launch on the last `console=` parameter defined on the ZBM KCL. Since a getty isn't spawned for anything, there's no reasonable way to launch it anywhere else except through SSH.

The kernel parameters `zbm.lines` and `zbm.columns` can be used to set the size of a serial interface. They aren't sanity checked; whatever integer value is set for each will be passed to `stty`.

**Comment by neilbags** [comment]
I also came here to ask about this. Is there any workaround available or enhancement planned that would support this?

**Reply by zdykstra** [reply]
There is no workaround short of embedding an SSH server in the ZBM image. There are currently no plans to add in support for multiple local controlling terminals.
